src/event.rs

```diff
@@ -32,6 +32,8 @@ impl Drop for Event {
     }
 }
 
+unsafe impl Send for Event {}
```

Comment:
`ExecuteKernel` is designed to be used as "builder" for the `clEnqueueNDRangeKernel` command.  
I cannot see a reason to pass an `ExecuteKernel` between threads, can you?